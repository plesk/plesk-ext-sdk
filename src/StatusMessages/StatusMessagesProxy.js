// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

export default class StatusMessagesProxy {
    toaster = null;

    setToaster = toaster => {
        this.toaster = toaster;
    };

    add(...args) {
        if (!this.toaster) {
            return false;
        }

        return this.toaster.add(...args);
    }

    update(...args) {
        if (!this.toaster) {
            return false;
        }

        return this.toaster.update(...args);
    }

    remove(...args) {
        if (!this.toaster) {
            return;
        }

        this.toaster.remove(...args);
    }

    clear() {
        if (!this.toaster) {
            return;
        }

        this.toaster.clear();
    }
}
