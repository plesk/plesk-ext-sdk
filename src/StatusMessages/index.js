// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

export { default } from './StatusMessages';
export { default as StatusMessagesProvider } from './StatusMessagesProvider';
export { default as StatusMessagesProxy } from './StatusMessagesProxy';
export { default as withStatusMessages } from './withStatusMessages';
