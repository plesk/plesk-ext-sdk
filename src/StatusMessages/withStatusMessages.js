// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import PropTypes from 'prop-types';
import { createElement } from 'react';

const withStatusMessages = Component => {
    const C = (props, context) => (
        <Component statusMessages={context.statusMessages} {...props} />
    );

    C.contextTypes = {
        statusMessages: PropTypes.object,
    };
    C.displayName = `withStatusMessages(${Component.displayName || Component.name})`;
    C.WrappedComponent = Component;

    return C;
};

export default withStatusMessages;
