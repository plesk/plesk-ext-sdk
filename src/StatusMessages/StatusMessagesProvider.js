// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { Component } from 'react';
import PropTypes from 'prop-types';
import StatusMessagesProxy from './StatusMessagesProxy';

export default class ConfigProvider extends Component {
    static propTypes = {
        children: PropTypes.node,
        value: PropTypes.object,
    };

    static childContextTypes = {
        statusMessages: PropTypes.object,
    };

    static defaultProps = {
        children: null,
        value: new StatusMessagesProxy(),
    };

    getChildContext() {
        return {
            statusMessages: this.props.value,
        };
    }

    render() {
        return this.props.children;
    }
}
