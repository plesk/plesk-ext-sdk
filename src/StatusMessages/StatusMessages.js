// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { PureComponent, Toaster, PropTypes } from '@plesk/ui-library';
import StatusMessagesProxy from './StatusMessagesProxy';
import StatusMessagesProvider from './StatusMessagesProvider';
import { withRouter } from 'react-router-dom';

export class StatusMessages extends PureComponent {
    static propTypes = {
        children: PropTypes.node.isRequired,
        location: PropTypes.shape({
            pathname: PropTypes.string.isRequired,
        }).isRequired,
        statusMessages: PropTypes.shape({
            clear: PropTypes.func.isRequired,
            setToaster: PropTypes.func.isRequired,
        }),
    };

    static defaultProps = {
        statusMessages: new StatusMessagesProxy(),
    };

    componentDidUpdate(prevProps) {
        if (prevProps.location.pathname !== this.props.location.pathname) {
            this.props.statusMessages.clear();
        }
    }

    render() {
        return (
            <StatusMessagesProvider value={this.props.statusMessages}>
                {this.props.children}
                <Toaster ref={this.props.statusMessages.setToaster} />
            </StatusMessagesProvider>
        );
    }
}

export default withRouter(StatusMessages);
