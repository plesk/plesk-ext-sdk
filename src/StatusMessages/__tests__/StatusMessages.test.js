// Copyright 1999-2018. Plesk International GmbH. All rights reserved.
import { createElement } from 'react';
import { StatusMessages } from '../StatusMessages';
import { shallow } from 'enzyme';

describe('StatusMessages', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <StatusMessages location={{ pathname: '/' }}>
                <div />
            </StatusMessages>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('clear messages', () => {
        const statusMessages = {
            clear: jest.fn(),
            setToaster: jest.fn(),
        };
        const wrapper = shallow(
            <StatusMessages location={{ pathname: '/' }} statusMessages={statusMessages}>
                <div />
            </StatusMessages>,
        );
        wrapper.setProps({ location: { pathname: '/next' } });

        expect(statusMessages.clear).toHaveBeenCalledWith();
    });
});
