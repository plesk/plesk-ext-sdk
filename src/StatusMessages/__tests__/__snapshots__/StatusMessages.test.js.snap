// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StatusMessages renders correctly 1`] = `
<ConfigProvider
  value={
    StatusMessagesProxy {
      "setToaster": [Function],
      "toaster": null,
    }
  }
>
  <div />
  <ForwardRef(Toaster) />
</ConfigProvider>
`;
